apps/obsidian/src/utils/templates.ts

```diff
@@ -86,11 +86,39 @@ export const applyTemplate = async ({
 
     const templateContent = await app.vault.read(templateFile);
 
-    const currentContent = await app.vault.read(targetFile);
+    const templateFrontmatter =
+      app.metadataCache.getFileCache(templateFile)?.frontmatter || {};
+    const currentFrontmatter =
+      app.metadataCache.getFileCache(targetFile)?.frontmatter || {};
+
+    const mergedFrontmatter = {
+      ...templateFrontmatter,
+      ...currentFrontmatter,
+    };
+
+    await app.fileManager.processFrontMatter(targetFile, (fm) => {
+      Object.assign(fm, mergedFrontmatter);
+    });
+
+    const normalizedContent = templateContent
+      .replace(/\r\n/g, "\n")
+      .replace(/\r/g, "\n");
+
+    const templateBodyMatch = normalizedContent.match(
```

Comment:
Let's use obsidian's [getFrontMatterInfo](https://docs.obsidian.md/Reference/TypeScript+API/getFrontMatterInfo) / [FrontMatterInfo](https://docs.obsidian.md/Reference/TypeScript+API/FrontMatterInfo) here to future proof for any changes, instead of normalizing and matching ourselves

something like 
```typescript
    const frontmatterInfo = getFrontMatterInfo(templateContent);
    let templateBody = "";

    if (frontmatterInfo.exists) {
      templateBody = templateContent.slice(frontmatterInfo.contentStart);
    } else {
      templateBody = templateContent;
    }
```